README.md

```diff
@@ -1,3 +1,262 @@
+|---------------|-----------------------------------|---------------------------------------------------------|
+| 2.12          | 2.4.8, 3.1.x, 3.2.x, 3.4.x, 3.5.x | Java, Scala, PySpark (`.py` files or Jupyter notebooks) |
+
+## Main features
+
+For a detailed list of features and how to use them, you can refer the official
+documentation
+[here](https://cloud.google.com/bigtable/docs/use-bigtable-spark-connector).
+A list of main features is as follows:
+
+### Catalog definition
+
+You can define a catalog as a JSON-formatted string, to convert from the
+DataFrame's schema to a
+format compatible with Bigtable. This is an example of a catalog JSON:
+
+```
+{
+  "table": {"name": "t1"},
+  "rowkey": "id_rowkey",
+  "columns": {
+    "id": {"cf": "rowkey", "col": "id_rowkey", "type": "string"},
+    "name": {"cf": "info", "col": "name", "type": "string"},
+    "birthYear": {"cf": "info", "col": "birth_year", "type": "long"},
+    "address": {"cf": "location", "col": "address", "type": "string"}
+  }
+}
+```
+
+Here, the columns `name`, `birthYear`, and `address` from the DataFrame are
+converted into Bigtable
+columns and the `id` column is used as the row key. Note that you could also
+specify *compound* row keys,
+which are created by concatenating multiple DataFrame columns together.
+
+### Writing to Bigtable
+
+You can use the `dataFrame.write()` format write to Bigtable.
+This is a sample snippet of writing to Bigtable using Java:
+
+```java
+Dataset<Row> dataFrame;
+// Adding some values to dataFrame.
+dataFrame
+  .write()
+  .format("bigtable")
+  .option("catalog", catalog)
+  .option("spark.bigtable.project.id", projectId)
+  .option("spark.bigtable.instance.id", instanceId);
+```
+
+### Reading from Bigtable
+
+You can use the `spark.read()` format to read from Bigtable.
+This is a sample snippet of reading from Bigtable using Java:
+
+```java
+Dataset<Row> dataFrame = spark
+    .read()
+    .format("bigtable")
+    .option("catalog", catalog)
+    .option("spark.bigtable.project.id", projectId)
+    .option("spark.bigtable.instance.id", instanceId)
+    .load();
+```
+
+### Runtime configurations
+
+You can use `.option(<config_name>, <config_value>)` in Spark to pass different
+runtime configurations to
+the connector. For example, Bigtable project and instance ID or settings for
+timestamp and timeout configurations.
+For a full list of configurations, refer to
+`spark-bigtable_2.12/src/main/scala/com/google/cloud/spark/bigtable/datasources/BigtableSparkConf.scala`,
+where these configs are defined.
+
+### Bigtable emulator support
+
+When using the connector locally, you can start a Bigtable emulator server and
+set the environment variable
+`export BIGTABLE_EMULATOR_HOST=localhost:<emulator_port>` to use the connector
+with the emulator instead of a
+real Bigtable instance. You can refer to the
+[Bigtable emulator documentations](https://cloud.google.com/bigtable/docs/emulator)
+for more details on using it.
```

Comment:
nit, When using the connector locally, you can start a Bigtable emulator server and
set the environment variable `export BIGTABLE_EMULATOR_HOST=localhost:<emulator_port>` in the same environment where Spark is launched. The connector will use the emulator instead of a real Bigtable instance.